hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaSystemManagement.java

```diff
@@ -223,11 +223,17 @@ public TenantMetaData getTenantMetadata(final String tenant) {
      * @return the initial created {@link TenantMetaData}
      */
     private TenantMetaData createInitialTenantMetaData(final String tenant) {
-        return systemSecurityContext.runAsSystemAsTenant(
-                () -> DeploymentHelper.runInNewTransaction(txManager, "initial-tenant-creation", status -> {
+        return systemSecurityContext.runAsSystemAsTenant(() -> {
+            try {
+                return DeploymentHelper.runInNewTransaction(txManager, "initial-tenant-creation", status -> {
                     final DistributionSetType defaultDsType = createStandardSoftwareDataSetup();
                     return tenantMetaDataRepository.save(new JpaTenantMetaData(defaultDsType, tenant));
-                }), tenant);
+                });
+            } catch (final TransactionExecutionException e) {
```

Comment:
I don't think it is a good idea to intercept the `TransactionExecutionException` here. Or at least we need to rethrow it rather than returning null, because it can just lead to NullPointer exceptions instead (see e.g. AmqpMessageDispatcherService#convertArtifact or JpaDistributionSetManagement#create)